halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/config/v1/ResourceConfig.java

```diff
@@ -17,13 +17,49 @@
 package com.netflix.spinnaker.halyard.config.config.v1;
 
 import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.context.annotation.Bean;
 import org.springframework.stereotype.Component;
 import org.yaml.snakeyaml.DumperOptions;
 import org.yaml.snakeyaml.Yaml;
 
 @Component
 public class ResourceConfig {
+  /**
+   * Path to where the halconfig file is located.
+   *
+   * @param path Defaults to ~/.hal/config.
+   * @return The path with home (~) expanded.
+   */
+  @Bean
+  String halconfigPath(@Value("${halconfig.filesystem.halconfig:~/.hal/config}") String path) {
+    return replaceBashHome(path);
+  }
+
+  /**
+   * Version of halyard.
+   *
+   * This is useful for implementing breaking version changes in Spinnaker that need to be migrated by some tool
+   * (in this case Halyard).
+   *
+   * @param version is the version (seems like the i.d. function for Spring Boot).
+   * @return the version of halyard.
+   */
+  @Bean
+  String halyardVersion(@Value("${Implementation-Version:unknown}") String version) {
```

Comment:
Implementation-Version is not currently universal. I think only Clouddriver and Front50 set it.

For stackdriver I used getClass().getPackage().getImplementationVersion(). That will give you the version of the thing that this is packaged in (e.g. the halyard version) as apposed to the thing that this is deployed in (e.g. clouddriver version if this were embedded at runtime, for example).